install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -61,8 +61,39 @@ spec:
           imagePullPolicy: {{ .Values.nodeinit.image.pullPolicy }}
           securityContext:
             privileged: true
-{{- if .Values.nodeinit.revertReconfigureKubelet }}
           lifecycle:
+{{- if and (.Values.eni.enabled) (not .Values.enableIPv4Masquerade) }}
```

Comment:
Why is the check for `not .Value.enableIPv4Masquerade` needed?

Reply:
The original issue suggests this is only necessary when masquerading is disabled — and that was also @aanm's understanding when we discussed this, so we agreed on the need for this check.

Reply:
My understanding is that the problem is broadly to do with left-over state, and that it generally needs to cleaned up. The fact that this state affects Cilium when masquerading is enabled seems rather secondary to me.